src/Licensing/SiteLicense.php

```diff
@@ -11,6 +11,11 @@ public function key()
         return config('statamic.system.license_key');
     }
 
+    public function usesLegacyKey()
+    {
+        return ! preg_match('/^[a-zA-Z0-9]{16}$/', $this->key());
```

Comment:
You could probably use the `Statamic\Facades\Pattern::isUuid()` helper method right here.

Reply:
Sorry, completely missed the notification for this. Yeah, could use this helper, thanks 🙏 But this made me think about that it could be even better to rename the method from `usesLegacyKey` to maybe `usesIncorrectKeyFormat` so more use case are being covered? (missing character or else) Then the regular expression would stay in place, I guess? The message in the CP could then be clearer, like `Using legacy key or missing a character. Please check the key.` Thoughts?

Reply:
Sure, that works!

Reply:
Awesome, adjusted the code accordingly!